### PR TITLE
In 2D conv layers, use padding='same' rather than padding layer when the case is verified

### PR DESCRIPTION
**Pull request by talcs**
What I did was:

- Changed the hardcoded instances of `padding='valid'` and `padding='VALID'` to a variable called `padding_mode`, assigned with `'VALID'` (Keras is case-insensitive. TF might be sensitive).

- Skipping the 2D padding layer and setting `padding_mode='SAME'` if and only if:
    - the padding equals to the conv kernel dimension divided by 2 and floored (e.g. padding=1 when size = 3, padding = 2 when size = 5, padding = 3 when size = 7, etc.) 
   - the conv kernel has odd dimensions (since for an even-sized kernel, it is not well defined where the padding should be added for a "same" padding)

I hope this is an acceptable change from your side.

### CODE REVIEW

**Comment by talcs** [comment]
There's a problem with my patch. It does not behave the same. Without the patch, the outputs are identical to those produced by Pytorch while my patch breaks it.
